src/db/sysdb_ops.c

```diff
@@ -5625,3 +5637,170 @@ int sysdb_invalidate_cache_entry(struct sss_domain_info *domain,
     talloc_zfree(tmp_ctx);
     return ret;
 }
+
+/* === Operation On Indexes ================================== */
+errno_t sysdb_ldb_list_indexes(TALLOC_CTX *mem_ctx,
+                               struct ldb_context *ldb,
+                               const char *attribute,
+                               const char ***_indexes)
```

Comment:
I usually NULLify the output parameters - just to avoid surprises. It is matter of preference, I checked our code and somewhere we do it, somewhere we do not. Feel free to ignore this comment.

Reply:
Most SSSD functions I've seen do not touch the return values (keeping what the user provided) until making sure the function succeeds. Callers can always nullify the pointer themselves before the call to the function.
I though that was a global policy and I abided by it. I won't modify this here, but I'll keep it in mind for the future.

Reply:
General habit in our code base is to not touch the output variables unless the function is successful. But it is about author preference.